tmk_core/common/action.c

```diff
@@ -886,10 +886,19 @@ void clear_keyboard(void)
  * FIXME: Needs documentation.
  */
 void clear_keyboard_but_mods(void)
+{
+    clear_keys();
+    clear_keyboard_but_mods_and_keys();
+}
+
+/** \brief Utilities for actions. (FIXME: Needs better description)
+ *
+ * FIXME: Needs documentation.
+ */
+void clear_keyboard_but_mods_and_keys()
 {
     clear_weak_mods();
```

Comment:
Maybe we should add: 
```suggestion
#ifndef STRICT_MOD_LAYER_RELEASE
    clear_weak_mods();
#endif
```

Or even just remove it:
```suggestion
```

Reply:
Looking at the code, I don't think there is a risk here.  

It appears that weak mods are only used by: 
* Modifier'ed keys
* Mod Tap
* Tap Dance

Everything else uses real mods. 

Additionally, I found a bug/unexpected behavior (L203 of action.c).  It clears all weak mods, instead of just releasing just the correct one.  Not sure how to fix that though.

Reply:
Also, since `clear_keyboard_but_mods_and_keys` is only used a couple of places, we could a a bool to it's parameters, and then clear the mods based on that.

(mostly just spitballing here)